doc/design/game_design_document.md

```diff
@@ -0,0 +1,165 @@
+  unique effects and strategies.
+- **PvPvE**: Players compete against each other and AI-controlled bots, adding
+  an extra layer of challenge.
+- **Interactive environment**: The game world includes destructible objects and
+  other interactable elements that players can use to their advantage.
+
+### Game flow
+
+1. **Matchmaking**: Players join a lobby and are matched with others based on
+   skill level.
+2. **Character selection**: Players choose or create their characters, selecting
+   their class and skills.
+3. **Gameplay**: Players engage in combat, using their skills strategically to
+   defeat opponents and interact with the environment.
+4. **End-game**: Statistics are displayed at the end of each match, showing
+   player performance and other metrics.
+
+### Controls
+
+- **Keyboard and mouse**: Primary control method for PC.
+- **Optional controller support**: Future updates may include support for game controllers.
+
+## Game world
+
+### Setting and theme
+
+Project OMG is set in a fantastical world with various themed arenas. Each
+arena offers unique challenges and opportunities for interaction.
+
+### Levels and environments
+
+The game features multiple arenas, each with destructible objects and
+interactive elements that can be used strategically during matches.
+
+## Characters
+
+### Player characters
+
+- **Classes**: Various classes with unique skills and abilities.
+- **Skills**: Each class has a set of skills that can be combined for synergy effects.
+
+### Non-player characters (NPCs)
+
+- **Bots**: Intelligent AI-controlled bots that players will compete against in
+  addition to other players.
+
+## Story
+
+The game is match-based with no overarching story progression. Each match is a
+standalone experience focusing on the core gameplay mechanics and interactions.
+
+## User interface (UI)
+
+### Main menu
+
+- **Lobby**: Join or create a match, view statistics, and manage character settings.
+
+### In-game HUD
+
+- **Health and mana Bars**: Display the player's current health and mana.
+- **Skill cooldowns**: Show the cooldown status of the player's skills.
+- **Match timer**: Indicates the remaining time in the match.
+
+### Other interfaces
+
+- **End-game statistics**: Detailed statistics displayed at the end of each match.
+
+## Technical specifications
+
+### Software architecture
+
+A monolithic architecture will be used for simplicity, ensuring all components
+are integrated into a single codebase.
+
+### Technology stack
+
+- **Programming language**: Python
+- **Game engine**: Pygame or a similar Python-based game engine
+- **Networking**: WebSockets for real-time communication
+- **Database**: TBD
+
+### Performance requirements
+
+- **Frame rate**: Targeting 60 FPS for smooth gameplay.
+- **Latency**: Low-latency communication to ensure real-time interactions.
+
+## Art and sound
+
+### Art style
+
+A stylized 2D art style with vibrant colors and detailed environments.
```

Comment:
Lets give an example of an actual game, so that the desired art style would be understood by everyone.
For example, children of mortar. (this is just a suggestion, we should discuss)

Reply:
I think this is too detailed for GDD.